### PR TITLE
Virtual inventory notification on shipment reverse or reactivate

### PR DESCRIPTION
**Pull request by mrie04**

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java [hunk 1]
```diff
@@ -104,11 +124,7 @@ public void init()
 	@DocValidate(timings = { ModelValidator.TIMING_AFTER_REVERSECORRECT, ModelValidator.TIMING_AFTER_REVERSEACCRUAL })
 	public void destroyHandlingUnitsForReceipt(final I_M_InOut inout)
 	{
-		if (inout.isSOTrx())
-		{
-			// TODO: change HUStatus from Shipped back to Picked
-		}
-		else
+		if (!inout.isSOTrx())
```
Comment (mrie04) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java:127]: Already in de.metas.handlingunits.model.validator.M_InOut#afterReactivate

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java [hunk 2]
```diff
@@ -415,4 +421,36 @@ public void validateAttributesOnShipmentCompletion(final I_M_InOut shipment)
...
+		final List<UserNotificationRequest> notificationRequestList = new ArrayList<>();
+		for (final InventoryId inventoryId : inventoryIds)
+		{
+			notificationRequestList.add(UserNotificationRequest.builder()
+												.recipientUserId(Env.getLoggedUserId())
+												.contentADMessage(MSG_SHIPMENT_WITH_VIRTUAL_HU_INVENTORY)
+												.contentADMessageParam(shipment.getM_InOut_ID())
+												.targetAction(UserNotificationRequest.TargetRecordAction.ofRecordAndWindow(
+														TableRecordReference.of(I_M_Inventory.Table_Name, inventoryId),
+														adWindowDAO.retrieveOverrideWindowIdOrWindowId(AdWindowId.ofRepoId(168))))
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java:451]: why do we have to specify the windowId?

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java [hunk 3]
```diff
@@ -140,4 +140,7 @@ public interface IADWindowDAO extends ISingletonService
 	ImmutableSet<AdWindowId> retrieveAllActiveAdWindowIds();
 
 	Optional<QuickInputConfigLayout> getQuickInputConfigLayout(AdTabId adTabId);
+
+	@NonNull
+	AdWindowId retrieveOverrideWindowIdOrWindowId(@NonNull final AdWindowId windowId);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java:145]: pls remove this. if u really need it then pls use de.metas.document.references.zoom_into.SqlCustomizedWindowInfoMapRepository#get

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java (hunk 2 above)
Comment (teosarca) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_InOut.java:451]: (as discussed)

**Review by teosarca — APPROVED** [review]

**Pull request merged**